packages/core/src/libraries/quota.ts

```diff
@@ -6,35 +8,70 @@ import { getTenantSubscriptionPlan } from '#src/utils/subscription/index.js';
 import { type FeatureQuota } from '#src/utils/subscription/types.js';
 
 import { type CloudConnectionLibrary } from './cloud-connection.js';
+import { type ConnectorLibrary } from './connector.js';
 
 export type QuotaLibrary = ReturnType<typeof createQuotaLibrary>;
 
-export const createQuotaLibrary = (queries: Queries, cloudConnection: CloudConnectionLibrary) => {
+const notNumber = (): never => {
+  throw new Error('Only support usage query for numberic quota');
+};
+
+export const createQuotaLibrary = (
+  queries: Queries,
+  cloudConnection: CloudConnectionLibrary,
+  connectorLibraray: ConnectorLibrary
+) => {
   const {
     applications: { countNonM2MApplications, countM2MApplications },
     resources: { findTotalNumberOfResources },
+    hooks: { getTotalNumberOfHooks },
+    roles: { countRoles },
+    scopes: { countScopesByResourceId },
+    rolesScopes: { countRolesScopesByRoleId },
   } = queries;
 
-  const getTenantUsage = async (key: keyof FeatureQuota): Promise<number> => {
-    if (key === 'applicationsLimit') {
-      return countNonM2MApplications();
-    }
-
-    if (key === 'machineToMachineLimit') {
-      return countM2MApplications();
-    }
-
-    if (key === 'resourcesLimit') {
-      const { count } = await findTotalNumberOfResources();
-      return count;
-    }
+  const { getLogtoConnectors } = connectorLibraray;
+
+  const tenantUsageQueries: Record<
+    keyof FeatureQuota,
+    (queryKey?: string) => Promise<{ count: number }>
+  > = {
+    applicationsLimit: countNonM2MApplications,
+    hooksLimit: getTotalNumberOfHooks,
+    machineToMachineLimit: countM2MApplications,
+    resourcesLimit: findTotalNumberOfResources,
```

Comment:
We should exclude the default management API.

Reply:
rebased